### DISCUSSION TITLE
Testing Input System library update

### DISCUSSION THREAD

**Discussion by Frooxius** [Yellow-Dog-Man/Resonite-Issues#3174] [Pre-release Testing]
Hello everyone!

I got another smaller thing for you to test!

## What is being tested
I've updated the Input System library that's used on Unity side to latest version, from 1.0.2.0 to 1.11.2.0. This is quite a big jump and I want to make sure this doesn't break anything weirdly.

Additionally there's some possibility this might fix crashes when switching between controllers & hand trackers with Quest headsets with Steam Link: https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/3066

## Current build number
2024.11.12.1329 - same as current public version (please double check that the prerelease has downloaded)

## How to test
- On Steam, switch to the `prerelease` branch and make sure the build updates before you run it
- Make sure that Steam downloads the new build

You can test alone, there's no multiplayer parts that should be affected.

### What to test
- Test in desktop & VR with variety of input methods
- Test with keyboard, mouse, gamepad
- In VR, test with different controllers if you have access to them
- If you have Quest headset and Steam Link, try switching between hand tracking and controllers back and forth to see if crash occurs

### What NOT to test & report
- Any pre-existing bugs. If you're unsure, switch back to main build and test it again. Report it *only* if it occurs on prerelease (otherwise just make a normal bug report)

## If you find something that got broken
- Create a GitHub issue for a pre-release bug: https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/new?assignees=shiftyscales&labels=prerelease&projects=&template=prerelease.yml
- Make sure to include the log
- Please include description of what happened. Be succinct, but exact.

## If you find that stuff works okay
Please report it too and include details on what you generally tested, so we know things are okay. You don't need to make a new issue for this - just a commend in this Discussion is fine.

This lets us know that a bunch of stuff was tested and get better confidence that stuff won't break with release.

## How long will testing go for?
With the nature of these changes issues should become apparent pretty quickly - at least the frequent ones. If we get a bunch of reports of things working okay, it'll be released within a few days. If there's blocker issues cropping up, we'll either hold up the release or not make the upgrade (if it doesn't resolve the crashing issue).

**Comment by ModernBalloonie** [comment]
Unfortunately the unity crash still happens when going to the quest menu when hand tracking is initalized in SteamVR. I was just able to reproduce it again.

**Comment by ko-tengu** [comment]
Yeah, zero changes on my end, everything that worked continues to work, switching controllers continues to crash.
[Player.log](https://github.com/user-attachments/files/17728740/Player.log)
[BUFU - 2024.11.12.1329 - 2024-11-13 03_06_53.log](https://github.com/user-attachments/files/17728732/BUFU.-.2024.11.12.1329.-.2024-11-13.03_06_53.log)
[crash.dmp](https://github.com/user-attachments/files/17728745/crash.dmp)
[error.log](https://github.com/user-attachments/files/17728746/error.log)

keyboard and mouse still works, xbox controller works, vr switching doesn't lock any inputs in an unexpected way

**Comment by bredo228** [comment]
Seems to mostly work on desktop for me, ran into an odd issue with the freecam though (https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/3175)

**Comment by ohzee00** [comment]
Tested desktop and VR, desktop works fine with protoflux nodes that work based off of keyboard inputs working fine.

VR was tested with a version of steamlink that allows handtracking on a Quest Pro with index controllers. Everything was working correctly including gestures, however on a third switch from index controllers to handtracking (that I turn off the controllers completely and have to "activate" handtracking by opening the oculus dash) I got a unity crash when pinching my finger on a contacts page.
[DESKTOP-V75BHJO - 2024.11.12.1329 - 2024-11-13 03_13_00.log](https://github.com/user-attachments/files/17729040/DESKTOP-V75BHJO.-.2024.11.12.1329.-.2024-11-13.03_13_00.log)

[crash.dmp](https://github.com/user-attachments/files/17729047/crash.dmp)
[error.log](https://github.com/user-attachments/files/17729048/error.log)
[Player.log](https://github.com/user-attachments/files/17729049/Player.log)

So basically same behavior as release, at least I haven't seen anything strange myself in the brief time I tested.

**Comment by Talakai** [comment]
I noticed no difference in desktop with mouse, keyboard or controller, but I did find it still unable to support mousekeys.

However when i got to desktop mode was when i ran into a problem. The game kept moving me to the left, even after a reboot. I tried disconnecting the controllers, reconnecting them. I tried moving to hands control mode and even desktop mode (f8) and i kept moving left. Somewhere between switching screens between steam and my resonite I got it to once stop moving to the left. But the moment i minimized resonite it kept moving to the left and didn't stop. Additionally i could not use my triggers to select anything, the game acted like both of them were busy at all times. I had to close the game through the f8 desktop mode both times i ran it.

Upon returning to the live version of the game the constant movement to the left stopped. I understand if there is difficulty understanding my explanation. I am willing and ready to repeat it anytime or do another test in person if anyone would like to evaluate or have me gather any particular log file or other settings that may help to understand the problem.

I used a quest pro, with steam link and quest pro controllers. I asked big red wolfy to try the same setup and he had his own unique problems but they were not related to my own. 

I will follow up with any information here on github but if anyone wants to reach out to me for a test please message me on discord through the discord id [Talakai]. We will move everything we find to here, but that will be much easier for me to set up a time to interact with someone who would be more informed on how to properly report my findings. 

Edit: I am trying to use github more but I am still very new to this, and to understanding how to do QA tests. Forgive my inexperience and please have patience with me. I very strongly want to do my part to help this program and community, there will just be a learning period.

**Comment by Krzeszny** [comment]
Tested switching between desktop and VR (Reverb G2) and 2 pairs of controllers, switching from WMR controllers to Reverb G2 controllers. Nothing extraordinary.

While I can't switch back from the second pair to the first pair, that's a main build bug that's always been a problem with SteamVR.

**Comment by Frooxius** [comment]
Thank you everyone for testing and reporting! 

Since this doesn't seem to resolve any issues (particularly the crashes) and there's a few new ones, I think we'll stick to the current version for the time being.

It was worth a shot, but we'll have to find other approaches to resolve the issues.

**Comment by Talakai** [comment]
Thank you for trying Froox. We'll get it soon enough~

I don't know if this helps but..
As someone who uses quest pro, and switches between hand and controller tracking. I do find that if i watch my packets qued, look away from everyone and anything that moves, then tap the two controllers together I won't crash. It only seems to consistently crash if it's trying to do other thinking, like loading an object or calculate for people's avatars moving.

**Reply by ko-tengu** [reply]
in my experience (and a couple of others i've talked to) it's simply the third time it has to handle a switch.

**Reply by LexiBasilisk** [reply]
@Talakai @ko-tengu - I'd suggest further feedback regarding the hand tracking Unity crash be posted on the original issue #3066 instead.
